src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/HalfConverter.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a type converter to convert half-precision, floating point number objects to and
+    /// from various other representations.
+    /// </summary>
+    public class HalfConverter : BaseNumberConverter
+    {
+        /// <summary>
+        /// Determines whether this editor will attempt to convert hex (0x or #) strings
+        /// </summary>
+        internal override bool AllowHex => false;
+
+        /// <summary>
+        /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
+        /// </summary>
+        internal override Type TargetType => typeof(Half);
+
+        /// <summary>
+        /// Convert the given value to a string using the given radix
+        /// </summary>
+        internal override object FromString(string value, int radix) => value is null ? (object)0 : Half.Parse(value, CultureInfo.CurrentCulture);
+
+        /// <summary>
+        /// Convert the given value to a string using the given formatInfo
+        /// </summary>
+        internal override object FromString(string value, NumberFormatInfo? formatInfo) => Half.Parse(value, NumberStyles.Float, formatInfo);
+
+        /// <summary>
+        /// Convert the given value from a string using the given formatInfo
+        /// </summary>
+        internal override string ToString(object value, NumberFormatInfo? formatInfo)
+        {
+            return ((Half)value).ToString("R", formatInfo);
```

Comment:
This is .NET 7 exclusive so we can just do `ToString(formatInfo)`. We return the shortest round-trippable string by default on .NET Core 3.1 and later.